### PR TITLE
Add Remove Todo

### PR DESCRIPTION
**Pull request by MapleSyyrup**
- Add a remove feature
image

### CODE REVIEW

File: lib/models/router.dart [hunk 1]
```diff
@@ -3,9 +3,7 @@ import 'package:flutter/material.dart';
 import '../screens/todo_overview_screen.dart';
 import '../widgets/todo_items.dart';
 
-/* 
-Class for routers for easy reference 
-*/
+///Class for routers for easy reference
```
Comment (Nei1eveN) [inline lib/models/router.dart:6]: put this on the next line

File: lib/provider/todo_provider.dart [hunk 2]
```diff
@@ -13,27 +13,36 @@ addNewTodo = function to add a task in the list based on the newDate
  else, a new task will be added*/
 
 class TodoProvider with ChangeNotifier {
-  Map<String, Todo> _todoList = {}; ///Map for _todolist
+  ///Map for _todolist
+  Map<String, Todo> _todoList = {};
```
Comment (Nei1eveN) [inline lib/provider/todo_provider.dart:17]: make this private

File: lib/provider/todo_provider.dart [hunk 3]
```diff
@@ -13,27 +13,36 @@ addNewTodo = function to add a task in the list based on the newDate
...
   Map<String, Todo> get todoList {
-    return {..._todoList}; /// getter of _todoList
+    return {..._todoList};
   }
 
-  int get todoCount => _todoList.length; ///count of the _todoList
+  ///count of the _todoList
+  int get todoCount => _todoList.length;
 
   void addNewTodo(String newTodo, String newDate) {
```
Comment (Nei1eveN) [inline lib/provider/todo_provider.dart:30]: add a document above this line

File: lib/provider/todo_provider.dart [hunk 4]
```diff
@@ -13,27 +13,36 @@ addNewTodo = function to add a task in the list based on the newDate
...
+          date: newDate,
+        );
+      });
     }
-    notifyListeners(); ///notifyListeners notify the provider for any changes in the map
+    notifyListeners();
+  }
+
+  ///Removes a task
+  void removeItem(String todoDate) {
```
Comment (Nei1eveN) [inline lib/provider/todo_provider.dart:48]: explain the purpose of passed parameter

File: lib/provider/todo_provider.dart (hunk 3 above)
Comment (Nei1eveN) [reply lib/provider/todo_provider.dart:30]: explain the passed parameters as well.

File: lib/screens/todo_overview_screen.dart [hunk 5]
```diff
@@ -57,23 +50,62 @@ class _TodoOverviewScreenState extends State<TodoOverviewScreen> {
               },
             )
           : ListView.builder(
-              ///if the map is not Empty, the list will be show
               itemCount: provider.todoList.length,
               itemBuilder: (ctx, i) {
                 final todo = provider.todoList.values.toList()[i];
-                return TodoItems(todo: todo);
+                return Dismissible(
```
Comment (Nei1eveN) [inline lib/screens/todo_overview_screen.dart]: create a separate widget class for this instead

File: lib/widgets/todo_items.dart [hunk 6]
```diff
@@ -23,7 +23,9 @@ class TodoItems extends StatefulWidget {
 }
 
 class _TodoItemsState extends State<TodoItems> {
-  bool _isCompleted = false; ///this is used for the Checkbox
+  bool _isCompleted = false;
+
+  ///this is used for the Checkbox
```
Comment (Nei1eveN) [inline lib/widgets/todo_items.dart]: ?

File: lib/widgets/todo_items.dart [hunk 7]
```diff
@@ -38,42 +40,41 @@ class _TodoItemsState extends State<TodoItems> {
...
+          onPressed: () {
+            return showModalBottomSheet(
+                isScrollControlled: true,
+                context: context,
+                builder: (BuildContext context) {
+                  return GestureDetector(
+                    onTap: () {},
+                    child: AddTodo(
+                      addTodo: (String newTodo, String newDate) => provider.addNewTodo(newTodo, newDate),
+                      task: widget.todo.task,
```
Comment (Nei1eveN) [inline lib/widgets/todo_items.dart]: `widget.todo` DRY

**Review by Nei1eveN — CHANGES_REQUESTED** [review]

File: lib/provider/todo_provider.dart (hunk 2 above)
Comment (MapleSyyrup) [reply lib/provider/todo_provider.dart:17]: I think this is already private because of underscore"_".

Comment (Nei1eveN) [reply lib/provider/todo_provider.dart:17]: I see.

File: lib/widgets/dismissible.dart [hunk 8]
```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../provider/todo_provider.dart';
+import '../models/todo.dart';
+import 'todo_items.dart';
+
+
+class BuildDismissible extends StatelessWidget {
```
Comment (Nei1eveN) [inline lib/widgets/dismissible.dart:10]: Document what this is for

**Review by Nei1eveN — CHANGES_REQUESTED** [review]

**Review by Nei1eveN — APPROVED** [review]

**Pull request merged**